src/psyclone/tests/psyir/frontend/fparser2_where_handler_test.py

```diff
@@ -511,3 +528,38 @@ def test_where_ordering(parser):
     assert isinstance(result[1], Loop)
     assert isinstance(result[2], Call)
     assert isinstance(result[3], Loop)
+
+
+@pytest.mark.parametrize(
+    "code, size_arg",
+    [("where (my_type%var(:) > epsi20)\n"
+      "my_type%array(:,jl) = 3.0\n", "my_type%var"),
+     ("where (my_type%block(jl)%var(:) > epsi20)\n"
+      "my_type%block(jl)%array(:,jl) = 3.0\n", "my_type%block(jl)%var")])
```

Comment:
I see we cover:
- StructureReference - ArrayMember
- StructureReference - ArrayOfStructureMember - ArrayMember

and CodeCov is happy with it but could you also add a case with:
- my_type(jl)%var(:)
- my_type%var%array(:)

So we also include: ArrayOfStructuresReference  and  Member in the mix.

Reply:
Bonus question (maybe in a separate test). Does a construct with ranges in different arrays work e.g. my_type(:)%var(:) . Is this even a valid where construct?

Reply:
Done.
As to your bonus question - it's not valid Fortran. This is checked in `_array_notation_rank` and tested in `fparser2_test.py`. I realised (while attempting to move this functionality into the PSyIR) that we should probably forbid this in the PSyIR too and I implemented that on my new branch `717_lift_where_restrictions`. Makes testing this very difficult :-)